src/controls/LevelingSensor/QskLevelingSensor.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <QskControl.h>
+#include <QskScaleTickmarks.h>
+#include <QVector3D>
+#include <QskAspect.h>
+
+#include <unordered_map>
+
+class QSK_EXPORT QskLevelingSensor : public QskControl
+{
+    Q_OBJECT
+        using Inherited = QskControl;
+public:
+    QSK_SUBCONTROLS(
+        OuterDisk,
+        Horizon,
+        TickmarksX,
+        TickmarksXLabels,
+        TickmarksY,
+        TickmarksYLabels,
+        TickmarksZ,
+        TickmarksZLabels)
+        using Tickmarks = QskScaleTickmarks;
+        using TickmarksLabels = QVector<QPair<qreal, QString>>;
+    explicit QskLevelingSensor(QQuickItem* parent = nullptr);
+public Q_SLOTS:
+    void setTickmarks(Qt::Axis axis, Tickmarks tickmarks);
+    void setTickmarksLabels(Qt::Axis axis, TickmarksLabels labels);
+    void setAngle(const QVector3D& degree);
+    void setAngle(Qt::Axis axis, float degree);
+
+    void setSubControlRotation(QskAspect::Subcontrol subControl, const QVector3D& degree);
+
+Q_SIGNALS:
+    void anglesChanged(const QVector3D& degree);
+    void subControlRotationChanged(QskAspect::Subcontrol subControl, const QVector3D& degree);
```

Comment:
maybe `virtual` as well as `virtual`getter

Reply:
Decision against virtual: setter and getter should not have more complexity and business logic that storing and returning values. Helper functions can be provided elsewhere.